Convert module to TypeScript

gzog opened the pull request:
**Description**
- Addresses: #2 
- Use same configuration found in latest version of [expect-ct](https://github.com/helmetjs/expect-ct/tree/21aa018565f96f4a8361c34e374ed4b817d0efcc)

gzog committed:
Convert module to TypeScript

EvanHahn commented on test/index.test.ts:
```diff
@@ -0,0 +1,47 @@
+import { IncomingMessage, ServerResponse } from 'http';
+
+import hidePoweredBy = require('..');
+
```
Nit: could you remove this line?

EvanHahn commented on package.json:
```diff
@@ -15,25 +15,44 @@
+  },
+  "main": "./dist/index.js",
+  "typings": "./dist/index.d.ts",
+  "files": [
```
Now that you've added this, could you `git rm .npmignore`?

EvanHahn reviewed (changes requested):
This is great—thanks so much for doing this!!

A few minor tweaks to make:

1. Could you remove the Standard.js badge from the readme, now that we're not using it?
2. Could you fill in the changelog or just remove it? And could you remove the reference to the changelog from the readme?

Again, this is super helpful—thanks for doing this!

gzog commented:
@EvanHahn Hey Evan. Thanks for feedback!
Made the following updates based on your comments:
- [x] Removed CHANGELOG.md
- [x] Removed reference to changelog in README.md
- [x] Removed reference to Standard.js badge in README.md
- [x] Removed redundant line in index.test.ts
- [x] Removed .npmignore

EvanHahn reviewed (approved):
Thanks so much for doing this! I'll merge and deploy shortly.

EvanHahn commented:
This has been released in `hide-powered-by@1.1.0`! This was a big help, thanks!!

Please add yourself to the [contributors list](https://github.com/helmetjs/helmetjs.github.io/blob/master/contributors/index.md) when you have a moment.
